### DISCUSSION TITLE
Receive midi messages from all connected devices simultaneously

### DISCUSSION THREAD

**Discussion by elmcapp** [djipco/webmidi#259] [Q&A]
Is there a way to receive midi messages from all connected midi devices simultaneously or do we still need to create a different listener for each midi devices.

**Comment by djipco** [comment]
No. Currently, you have to add a listener for all devices. Obviously, all those listeners can call the same function, though.

By the way, when you have questions such as this one, you should post to [GitHub Discussions](https://github.com/djipco/webmidi/discussions) instead of filing a new issue.

**Comment by elmcapp** [comment]
okay. Also can you provide an example of how to remove the listener

**Comment by djipco** [comment]
You can remove a listener from the `WebMidi` object by calling `WebMidi.removeListener()`. You can pass in options to control which listeners are removed. Check out the [documentation](https://webmidijs.org/api/classes/WebMidi#removeListener) for details.

You can also keep the listener in a variable and call its `remove()` method when you are done with it:

```js
const myListener = WebMidi.addListener("noteon", doSomethingThenRemoveListener);

doSomethingThenRemoveListener() {
    // do something
    myListener.remove()
}
```
